File: SDKs/amazonaws.com/redshift/2012-12-01/typescript/src/sdk/models/shared/scheduledactiontype.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { PauseClusterMessage } from "./pauseclustermessage";
import { ResizeClusterMessage } from "./resizeclustermessage";
import { ResumeClusterMessage } from "./resumeclustermessage";



// ScheduledActionType
/** 
 * The action type that specifies an Amazon Redshift API operation that is supported by the Amazon Redshift scheduler. 
**/
export class ScheduledActionType extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pauseCluster?: PauseClusterMessage;

  @SpeakeasyMetadata()
  resizeCluster?: ResizeClusterMessage;

  @SpeakeasyMetadata()
  resumeCluster?: ResumeClusterMessage;
}

File: SDKs/amazonaws.com/redshift/2012-12-01/python/src/sdk/models/shared/scheduledactiontype.py
from dataclasses import dataclass, field
from typing import Optional
from . import *


@dataclass
class ScheduledActionType:
    r"""ScheduledActionType
    The action type that specifies an Amazon Redshift API operation that is supported by the Amazon Redshift scheduler. 
    """
    
    pause_cluster: Optional[PauseClusterMessage] = field(default=None)
    resize_cluster: Optional[ResizeClusterMessage] = field(default=None)
    resume_cluster: Optional[ResumeClusterMessage] = field(default=None)
    
